pkg/model/model.go

```diff
@@ -527,6 +300,10 @@ func NewPipeline(source Source) Pipeline {
 	return SimplePipeline{source: source}
 }
 
+func (pipeline SimplePipeline) WithSource(source Source) Pipeline {
+	return SimplePipeline{source: source}
```

Comment:
original `pipeline`is lost

Reply:
A SimplePipeline is just composed of single source so i think there is nothing to save from the original pipeline